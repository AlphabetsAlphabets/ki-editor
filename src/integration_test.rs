#[cfg(test)]
mod integration_test {
    use std::time::Duration;

    use crossterm::event::Event;

    use crate::{
        canonicalized_path::CanonicalizedPath, key_event_parser::parse_key_events, screen::Screen,
    };

    // #[test]
    fn lsp_completion() {
        let mut screen = Screen::new().unwrap();
        let (sender, receiver) = std::sync::mpsc::channel();
        std::thread::sleep(Duration::from_secs(5));
        for event in parse_key_events("c-q").unwrap() {
            sender.send(Event::Key(event)).unwrap();
        }
        screen.run(None, receiver).unwrap();
    }
}
