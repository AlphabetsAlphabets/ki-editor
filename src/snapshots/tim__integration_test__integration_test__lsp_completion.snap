---
source: src/integration_test.rs
expression: test_runner.content()
---
<Untitled> [INSERT]                     │Completion [NORMAL:LINE]               
1│mod foo;                              │1│[1] option                           
2│use std::opt                          │2│[2] option_env                       
3│                                      │ │                                     
4│fn main() {                           │ │                                     
5│    foo::foo();                       │ │                                     
6│    println!("Hello, world!");        │ │                                     
7│}                                     │ │                                     
8│                                      │ │                                     
 │                                      │ │                                     
 │                                      │ │                                     
 │                                      │───────────────────────────────────────
 │                                      │ [NORMAL:CUSTOM]                       
 │                                      │  1│Optional values.                   
 │                                      │  2│                                   
 │                                      │  3│Type [`Option`] represents an optio
 │                                      │  4│is either [`Some`] and contains a v
 │                                      │  5│does not. [`Option`] types are very
 │                                      │  6│they have a number of uses:        
 │                                      │  7│                                   
 │                                      │  8│* Initial values                   
 │                                      │  9│* Return values for functions that 
 │                                      │ 10│  over their entire input range (pa
 │                                      │ 11│* Return value for otherwise report
